packages/schematics/src/command-line/workspace-schematic.ts

```diff
@@ -168,132 +135,42 @@ function executeSchematic(
- * for everything else.
- */
-class EngineHostHandlingWorkspaceSchematics implements EngineHost<any, any> {
-  readonly toolsHost: FileSystemEngineHost;
-  readonly defaultHost: NodeModulesEngineHost;
-
-  transformContext(): void {}
-
-  constructor(outDir: string) {
-    const transforms = validateOptionsWithSchema(
-      new CoreSchemaRegistry(standardFormats)
-    );
-    this.toolsHost = new FileSystemEngineHost(outDir);
-    this.toolsHost.registerOptionsTransform(transforms);
-
-    this.defaultHost = new NodeModulesEngineHost();
-    this.defaultHost.registerOptionsTransform(transforms);
-    this.defaultHost.registerTaskExecutor(BuiltinTaskExecutor.NodePackage, {
-      rootDirectory,
-      packageManager: fileExists('yarn.lock') ? 'yarn' : 'npm'
-    });
-  }
-
-  createCollectionDescription(name: string): CollectionDescription<any> {
-    return this.hostFor(name).createCollectionDescription(name);
-  }
-
-  createSchematicDescription(
-    name: string,
-    collection: CollectionDescription<any>
-  ): SchematicDescription<any, any> | null {
-    return this.hostFor(collection.name).createSchematicDescription(
-      name,
-      collection
-    );
-  }
-
-  getSchematicRuleFactory<OptionT extends object>(
-    schematic: SchematicDescription<any, any>,
-    collection: CollectionDescription<any>
-  ): RuleFactory<OptionT> {
-    return this.hostFor(collection.name).getSchematicRuleFactory(
-      schematic,
-      collection
-    );
-  }
-
-  createSourceFromUrl(url: Url, context: any): any {
-    return this.hostFor(context.schematic.collection.name).createSourceFromUrl(
-      url
-    );
-  }
-
-  transformOptions<OptionT extends object>(
-    schematic: SchematicDescription<any, any>,
-    options: OptionT
-  ): any {
-    return this.hostFor(schematic.collection.name).transformOptions(
-      schematic,
-      options
-    );
-  }
-
-  listSchematics(collection: any): string[] {
-    return this.listSchematicNames(collection.description);
-  }
-
-  listSchematicNames(collection: CollectionDescription<any>): string[] {
-    return this.hostFor(collection.name).listSchematicNames(collection);
-  }
-
-  createTaskExecutor(name: string): any {
-    return this.defaultHost.createTaskExecutor(name);
-  }
-
-  hasTaskExecutor(name: string): boolean {
-    return this.defaultHost.hasTaskExecutor(name);
-  }
+  const args = options._.slice(1);
+  workflow.registry.addSmartDefaultProvider('argv', (schema: JsonObject) => {
+    if ('index' in schema) {
+      return args[+schema.index];
+    } else {
+      return args;
+    }
+  });
+  delete options._;
+  const logger = createConsoleLogger(true, process.stdout, process.stderr);
 
-  private hostFor(name: string) {
-    return name === 'workspace-schematics' ? this.toolsHost : this.defaultHost;
+  try {
+    await workflow
+      .execute({
+        collection: path.join(outDir, 'workspace-schematics.json'),
```

Comment:
This is the key difference really. It's using the output path to define the collection.